[MAT-8271] Admin API: Overwrite Expected Values

jkotanchik-SB opened the pull request:
## MADiE PR

Jira Ticket: [MAT-8271](https://jira.cms.gov/browse/MAT-8271)
(Optional) Related Tickets:

### Summary

Admin endpoint for overwriting expected values on a Target Measure with data from a Source Measure.

Verifies the source and target are:
- In the same Measure Set.
  - AND
- Either:
        - Have the same Measure ID.
        - OR
        - Is the direct ancestor of the other. I.E., one is the draft of the versioned other.

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

jkotanchik-SB committed:
MAT-8271: Overwrite the expected values of the target Measure with the supplied Measure's expected values. Only verifies test case IDs match before overwriting.

Signed-off-by: Joseph Kotanchik <joseph.kotanchik@semanticbits.com>

jkotanchik-SB committed:
MAT-8271: Expand permissible source and targets to include same version value drafts and versioned measures.

Adds support for the following cases: Measure versioned pre-release and then drafted and modified post-release. Measure draft pre-release and then versioned, drafted, and modified post-release.

Signed-off-by: Joseph Kotanchik <joseph.kotanchik@semanticbits.com>

jkotanchik-SB committed:
MAT-8271: Change Admin measure version correction endpoint to avoid confusing with the new Admin correction endpoint.

Test tech debt: Replace deprecated MockBean with MockitoBean.

Signed-off-by: Joseph Kotanchik <joseph.kotanchik@semanticbits.com>

jkotanchik-SB committed:
MAT-8271: When using expected values from a versioned measure to correct its draft, the Test Case IDs will differ. Match on Test Case name and contents instead.

Correct expected value types whenever type does not  population basis (fhir) or patient basis (qdm).

Signed-off-by: Joseph Kotanchik <joseph.kotanchik@semanticbits.com>

jkotanchik-SB committed:
MAT-8271: Correct Test Case Group Ids to match target measure group ids. Needed when target and source are version and draft, respectively.

Signed-off-by: Joseph Kotanchik <joseph.kotanchik@semanticbits.com>

jkotanchik-SB committed:
MAT-8271: Simply draft/version check. Verify either source or target is the direct ancestor of the other by ensuring they have same measure set it and their version values match.

Signed-off-by: Joseph Kotanchik <joseph.kotanchik@semanticbits.com>

adongare commented on src/main/java/cms/gov/madie/measure/resources/AdminController.java:
```diff
@@ -264,6 +256,105 @@ public ResponseEntity<Measure> correctMeasureVersion(
+      throw new InvalidRequestException("Cannot overwrite differing measure versions.");
+    }
+
+    List<TestCase> targetTestCases = testCaseService.findTestCasesByMeasureId(id);
```
this may work
```suggestion
    List<TestCase> targetTestCases = targetMeasure.getTestCases();
```

jkotanchik-SB committed:
Simply target test case retrieval

Co-authored-by: Ashok Dongare <ashok.dongare@semanticbits.com>

jkotanchik-SB committed:
MAT-8271: Null check against target measure.

Signed-off-by: Joseph Kotanchik <joseph.kotanchik@semanticbits.com>

jkotanchik-SB committed:
MAT-8271: Overwrite any existing expected values with source content per BA discussion.

Ignore json content when matching source and target test cases as the user may have been updating the target test case.

Signed-off-by: Joseph Kotanchik <joseph.kotanchik@semanticbits.com>
